listen/client.py

```diff
@@ -90,11 +116,28 @@ def headers(self):
         :param int song_id: The id of the song you want to favourite
         :rtype: bool
         """
-        with aiohttp.ClientSession(headers=self._headers) as session:
-            async with session.post(SONG_FAVOURITES, data=json.dumps({"song": song_id})) as response:
-                boolean = await response.json()
-                return boolean["favorite"]
-
+        raise NotImplementedError()
+        # https://listen.moe/api/favorites/4726
+        # with aiohttp.ClientSession(headers=self._headers) as session:
+        #     async with session.post(SONG_FAVOURITES, data=json.dumps({"song": song_id})) as response:
+        #         boolean = await response.json()
+        #         return boolean["favorite"]
+    
+    @ensure_token
+    async def favorite(self, song_id: int, *, remove=False) -> None:
+        pass
+        async with aiohttp.ClientSession(headers=self._headers) as session:
+            method = session.post if not remove else session.delete
+            async with method("{}/api/favorites/{}".format(self._base_url, song_id)) as response:
+                # print(response.status)
+                # print(await response.json())
+                if response.status != 204:
+                    raise ListenError()
+    
+    @ensure_token
+    async def remove_favorite(self, song_id: int) -> None:
+        return await self.favorite(song_id, remove=True)
```

Comment:
If this is `self.remove_favorite` could be have a protected method like `self._favorite` and then change the current `self.favorite` to `self.add_favorite` (to keep naming consistency)